src/mlpack/bindings/markdown/print_docs.cpp

```diff
@@ -90,204 +139,237 @@ void PrintDocs(const std::string& bindingName,
+      paramsSet.clear();
+
+      cout << endl;
+
+      // Determine if there are any output options, to see if we need
+      // to print the header of the output options table.
+      bool hasOutputOptions = false;
+      map<string, ParamData>& parameters = params.Parameters();
+      for (map<string, ParamData>::iterator it = parameters.begin();
+          it != parameters.end(); ++it)
       {
-        if (!it->second.required)
+        if (!it->second.input)
         {
-          cout << ParamString(it->second.name) << " | ";
+          hasOutputOptions = true;
+          break;
         }
-        else
-        {
-          std::string name = ParamString(it->second.name);
-          name[1] = std::tolower(name[1]);
-          cout << name << " | ";
-        }
-      }
-      else
-      {
-        cout << ParamString(it->second.name) << " | ";
       }
-      cout << ParamType(it->second) << " | ";
-      string desc = boost::replace_all_copy(it->second.desc, "|", "\\|");
-      cout << desc; // just a string
-      // Print whether or not it's a "special" language-only parameter.
-      if (it->second.name == "copy_all_inputs" || it->second.name == "help" ||
-          it->second.name == "info" || it->second.name == "version")
+
+      if (hasOutputOptions)
       {
-        cout << "  <span class=\"special\">Only exists in "
-            << PrintLanguage(languages[i]) << " binding.</span>";
+        // Next, iterate through the list of output options.
+        cout << "### Output options" << endl;
+        cout << endl;
+        string outputInfo = PrintOutputOptionInfo();
+        if (outputInfo.size() > 0)
+          cout << outputInfo << endl;
+        cout << endl;
+        cout << "| ***name*** | ***type*** | ***description*** |" << endl;
+        cout << "|------------|------------|-------------------|" << endl;
       }
-      cout << " | ";
-      string def = PrintDefault(it->second.name);
-      if (def.size() > 0)
-        cout << "`" << def << "` |";
-      else
-        cout << " |";
+
+      PrintParamTable(bindingName, languages[i], params,
+          {"name", "type", "description"}, paramsSet,
+          false, false, false, true);
+      paramsSet.clear();
+
       cout << endl;
-    }
-    cout << endl;
 
-    // Determine if there are any output options, to see if we need
-    // to print the header of the output options table.
-    bool hasOutputOptions = false;
-    for (map<string, ParamData>::iterator it = parameters.begin();
-        it != parameters.end(); ++it)
-    {
-      if (!it->second.input)
+      cout << "### Detailed documentation" << endl;
+      cout << "{: #" << languages[i] << "_" << bindingName
+          << "_detailed-documentation }" << endl;
+      cout << endl;
+      string desc = replace_all_copy(doc.longDescription(),
+                                     "|", "\\|");
+      cout << desc << endl << endl;
+
+      if (doc.example.size() > 0)
+        cout << "### Example" << endl;
```

Comment:
Actually, there are no tags here for me to move around the `Example` section with.  I think maybe it would be easier if we chose to print the example earlier in `print_docs.cpp` if this is a wrapper.  Let me know what you think. :+1:

Reply:
Alternately, maybe it makes sense to keep the example where it is, but add an additional section that shows the constructor arguments right above the `Parameters` table.